<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>703</width>
    <height>397</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QString特性Demo</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Plain</enum>
      </property>
      <property name="lineWidth">
       <number>1</number>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>字符串截取</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QPushButton" name="btnFrontBack">
              <property name="text">
               <string>front / back</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnFirstLast">
              <property name="text">
               <string>first / last</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QPushButton" name="btnLeftRight">
              <property name="text">
               <string>left / right</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnSection">
              <property name="text">
               <string>section *</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>存储相关</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QPushButton" name="btnIsNullIsEmpty">
              <property name="text">
               <string>isNull / isEmpty</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnSizeLength">
              <property name="text">
               <string>size / length</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QPushButton" name="btnSize">
              <property name="text">
               <string>resize</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnFill">
              <property name="text">
               <string>fill</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>转换与修改</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QPushButton" name="btnTrimmed">
              <property name="text">
               <string>trimmed / simplified</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnRemove">
              <property name="text">
               <string>remove</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QPushButton" name="btnInsert">
              <property name="text">
               <string>insert</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnReplace">
              <property name="text">
               <string>replace</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QFrame" name="frame_2">
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Plain</enum>
      </property>
      <property name="lineWidth">
       <number>1</number>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_11">
         <item>
          <widget class="QGroupBox" name="groupBox_4">
           <property name="title">
            <string>输入</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_8">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="1,4">
              <item>
               <widget class="QLabel" name="label">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="text">
                 <string>字符串1：</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="comboBoxStr1">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="layoutDirection">
                 <enum>Qt::LayoutDirection::LeftToRight</enum>
                </property>
                <item>
                 <property name="text">
                  <string>D:\Users\Moe\AppData\Local</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>G:\Qt6Book\QtSamples</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>D:\Users\Moe\AppData\Local\Programs\ApiPost7\ffmpeg.dll</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Replaced str1.first(N+1) with str1.left(N+1) to get the first part of the string</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>字符串2：</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="comboBoxStr2">
                <property name="minimumSize">
                 <size>
                  <width>100</width>
                  <height>0</height>
                 </size>
                </property>
                <item>
                 <property name="text">
                  <string>\</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Qt</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>参数：</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBox"/>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_5">
           <property name="title">
            <string>结果显示</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_9">
              <item>
               <widget class="QPushButton" name="btnClear">
                <property name="text">
                 <string>清空文本</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPlainTextEdit" name="plainTextEdit"/>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
